src/app/meetups/add-meetups/time-mask.directive.ts

```diff
@@ -0,0 +1,26 @@
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+@Directive({
+  selector: 'input[planetTimeMask]'
+})
+export class TimeMaskDirective {
+
+  constructor(private _el: ElementRef) { }
+
+  @HostListener('input', [ '$event' ]) onInputChange(event) {
+    const inputValue = this._el.nativeElement.value;
+    let intValue = inputValue.replace(/[^0-9]*/g, '');
+    const regexp = new RegExp('^[0-2]$|^(0[0-9]|1[0-9]|2[0-3])([0-5]|[0-5][0-9])?$');
+    if (!regexp.test(intValue)) {
+      intValue = intValue.substring(0, intValue.length - 1);
```

Comment:
Check this for the case of the new input not being at the end of the string (i.e. '123' => '4123')